File: code/cpp/523.cpp
//https://leetcode.com/problems/continuous-subarray-sum/
bool checkSubarraySum(vector<int>& nums, int k) {
	unordered_map<int, int> mp;
	mp[0] = -1;
	size_t sum = 0, rem;
	for (int i = 0; i < nums.size(); i++) {
		sum += nums[i];
		rem = sum % k;
		if (mp.contains(rem)) {
			if (i - mp[rem] > 1)
				return true;
		}
		else
			mp[rem] = i;
	}
	return false;
}

File: code/python/523.py
#https://leetcode.com/problems/continuous-subarray-sum/
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mp = {0: -1}
        sum = 0
        for i in range(len(nums)):
            sum += nums[i]
            rem = sum % k
            if rem not in mp:
                mp[rem] = i
            elif i - mp[rem] > 1:
                return True
        return False
